Move transactions into the blocks table for performance, Py3 and PEP8 compliance, and configuration classes for different environments

ageis opened the pull request:
This is a million times faster! I moved both the transactions themselves (as a BLOB object) plus the `txs` = `len(json.loads(block['tx']))` as columns in the `blocks` table. I'm keeping the tx table around for now in case we'll still need it.

ageis committed:
Move transactions into the blocks table for performance

ageis committed:
Include more informative failure messages for search

ageis committed:
Store number of transactions as column in database

ageis committed:
Get rid of pagination query top/bottom height stuff

ageis committed:
Better Py3-compatible error messages

ageis committed:
More debug messages for validating input

ageis committed:
Enable Flask debugging options

ageis committed:
Still use tx table to find blocks by txid

ageis commented:
@arcalinea This one part is absolutely perplexing! Search is broken. I debugged every line in validate_input(), and it's ok. It seems to fail at render_template().
```
127.0.0.1 - - [04/May/2017 03:11:34] "GET / HTTP/1.1" 200 -
Searching: 00000003ff05987f4e1c1fb31c40a195fce2bf9c09c04a119dad685a114498c3
Search matches hexademical format of a block hash or txid.
Error: Failed to locate block by hash.
Error: Failed to locate block by txid.
Error: Failed to locate block by height.
```

Update it, I fixed it, the problem was the template contained `nextblockhash` instead of block['nextblockhash']. Everything is working fantastically.

ageis committed:
Remove unused imports per pyflakes

ageis committed:
Remove unused imports in loadblocks.py

ageis committed:
Auto-reload templates to suppress caching

ageis committed:
Use new transactions var in template + fix nextblockhash

ageis committed:
Remove unused imports in receiveblocks.py

ageis committed:
Retrive txs + blocks in get_single_block()

Remove caching code for now
Print more informative exceptions

ageis committed:
Remove size constraints on database fields

ageis committed:
Nitpick formatting changes

ageis committed:
Mostly adds configuration classes and PEP-compliant code

ageis committed:
Adds trivial newline

ageis committed:
Allow command-line arguments for loadblocks.py

arcalinea commented:
couldn't get this to run locally because of db file changes, but UTACK 

I'll merge my UI changes with this

ageis committed:
Massive improvements to Block Observatory code

Introduced function to check whether zcashd is running and RPC accessible.
Added argument parsing to loadblocks.py
Allows a range of blocks to be uploaded
